cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSparkPipelineDriver.java

```diff
@@ -161,6 +166,18 @@ public void run(DatasetContext context) throws Exception {
         sec.getRuntimeArguments().getOrDefault(Constants.CONSOLIDATE_STAGES, Boolean.TRUE.toString()));
       boolean shouldCacheFunctions = Boolean.parseBoolean(
         sec.getRuntimeArguments().getOrDefault(Constants.CACHE_FUNCTIONS, Boolean.TRUE.toString()));
+
+      // Initialize SQL engine instance if needed.
+      if (phaseSpec.getSqlEngineStageSpec() != null) {
+        String sqlEngineStage =
+          "sqlengine_" + Strings.nullToEmpty(phaseSpec.getSqlEngineStageSpec().getPlugin().getName()).toLowerCase();
+        Object sqlEngineInstance = pluginInstantiator.newPluginInstance(sqlEngineStage);
```

Comment:
@albertshau since the sqlEngineStage is not actually part of the `phaseSpec` for this pipeline, I wonder if you see any potential issues. I see that, internally, if the `PipelinePluginInstantiator` will delegate to the Spark Execution Context  in order to get the plugin instance, so this should initialize the plugin instance just fine.

Reply:
It isn't needed. The only thing that's needed after configuration is the ID that was assigned to the plugin at configure time.